static/js/modules/tables.js

```diff
@@ -396,127 +351,192 @@ function adjustFormHeight($table, $form) {
-        callbacks.mapQuery(api, data),
-        parsedFilters || {}
-      );
-      if (useHideNull) {
-        query = _.extend(
-          query,
-          {sort_hide_null: $hideNullWidget.is(':checked')}
-        );
-      }
-      query.sort = mapSort(data.order, columns);
-      $processing.show();
-      $.getJSON(
-        helpers.buildUrl(path, _.extend({}, query, baseQuery || {}))
-      ).done(function(response) {
-        callbacks.handleResponse(api, data, response);
-        callback(callbacks.preprocess(response));
-        callbacks.afterRender(api, data, response);
-        if (opts.hideEmpty) {
-          hideEmpty(api, data, response);
-        }
-      }).always(function() {
-        $processing.hide();
-      });
-    }
-  }, opts || {});
-  var api = $table.DataTable(opts);
-  callbacks = _.extend({
-    handleResponse: function() {},
-    afterRender: function() {}
-  }, callbacks);
-  if (useFilters) {
-    // Update filters and data table on navigation
-    $(window).on('popstate', function() {
-      filterPanel.filterSet.activate();
-      var tempFilters = filterPanel.filterSet.serialize();
-      if (!_.isEqual(tempFilters, parsedFilters)) {
-        api.ajax.reload();
-      }
-    });
+
+  if (this.opts.useFilters) {
+    $(window).on('popstate', this.handlePopState.bind(this));
   }
-  // Prepare loading message
-  $processing.hide();
-  $table.before($processing);
-  var $paging = $(api.table().container()).find('.results-info--top');
-  if (useHideNull) {
-    $paging.prepend($hideNullWidget);
+
+  this.$body.css('width', '100%');
+  this.$body.find('tbody').addClass('js-panel-toggle');
+}
+
+DataTable.prototype.destroy = function() {
+  this.api.destroy();
+  delete DataTable.registry[this.$body.attr('id')];
+};
+
+DataTable.prototype.handlePopState = function() {
+  this.filterSet.activate();
+  var filters = this.filterSet.serialize();
+  if (!_.isEqual(filters, this.filters)) {
+    this.api.ajax.reload();
   }
-  $table.css('width', '100%');
-  $table.find('tbody').addClass('js-panel-toggle');
-  if ($form) {
-    updateOnChange($form, api);
-    $table.on('draw.dt', adjustFormHeight.bind(null, $table, $form));
+};
+
+DataTable.prototype.ensureWidgets = function() {
+  if (this.hasWidgets) { return; }
+  this.$processing = $('<div class="overlay is-loading"></div>').hide();
+  this.$body.before(this.$processing);
+
+  if (this.opts.useHideNull) {
+    this.$hideNullWidget = $(hideNullTemplate());
+    var $paging = this.$body.closest('.dataTables_wrapper').find('.js-results-info');
+    $paging.prepend(this.$hideNullWidget);
   }
-}
+
+  this.hasWidgets = true;
+};
+
+DataTable.prototype.fetch = function(data, callback) {
+  var self = this;
+  self.ensureWidgets();
+  if (self.filterSet) {
+    pushQuery(self.filterSet.serialize(), self.filterSet.fields);
+    self.filters = self.filterSet.serialize();
+  }
+  var url = self.buildUrl(data);
+  self.$processing && self.$processing.show();
```

Comment:
JSHint didn't pick up on it because I told it not to. I don't mind short-circuit conditionals in simple cases like this--plus, the JSHint checker that would have complained here also catches lots of false positives with chai asserts. Anyway, I'm happy to use the standard conditional if if improves readability. Not a fan of single-line conditionals, though :stuck_out_tongue_winking_eye: